src/librustc_mir/borrow_check/nll/constraint_generation.rs

```diff
@@ -79,8 +88,12 @@ impl<'cx, 'gcx, 'tcx> ConstraintGeneration<'cx, 'gcx, 'tcx> {
                 .drop
                 .simulate_block(self.mir, bb, |location, live_locals| {
                     for live_local in live_locals.iter() {
-                        let live_local_ty = self.mir.local_decls[live_local].ty;
-                        self.add_drop_live_constraint(live_local_ty, location);
+                        if let LookupResult::Exact(mpi) = self.move_data.rev_lookup.find(&Lvalue::Local(live_local)) {
```

Comment:
@nikomatsakis this should always be true. Your comment in #45665 suggested that `move_data.move_path_index` was somehow available here, but I'm not sure if it's missing something (looks cut off in the comment there) or only worked in earlier versions of the code. It's only available on MoveOutIndex, and I'm not sure how to get this through a local. Should I create a method on `MovePathLookup` to avoid the if check, or is there something existing that I could have used and just missed?